gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/event/CountEventBuilder.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.metrics.event;
+
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import org.apache.gobblin.metrics.GobblinTrackingEvent;
+
+/**
+ * The builder builds builds a specific {@link GobblinTrackingEvent} whose metadata has
+ * {@value GobblinEventBuilder#EVENT_TYPE} to be {@value #COUNT_EVENT_TYPE}
+ *
+ * <p>
+ * Note: A {@link CountEventBuilder} instance is not reusable
+ */
+public class CountEventBuilder extends GobblinEventBuilder {
+
+  public static final String COUNT_EVENT_TYPE = "CountEvent";
+  public static final String COUNT_KEY = "count";
+  @Setter
+  @Getter
+  private int count;
+
+  public CountEventBuilder(String name) {
+    this(name, NAMESPACE);
+  }
+
+  public CountEventBuilder(String name, String namespace) {
+    super(name, namespace);
+    this.metadata.put(EVENT_TYPE, COUNT_EVENT_TYPE);
+  }
+
+  /**
+   *
+   * @return {@link GobblinTrackingEvent}
+   */
+  @Override
+  public GobblinTrackingEvent build() {
+    this.metadata.put(COUNT_KEY, Integer.toString(count));
```

Comment:
As implemented right now, it is easy for the user to forget to set the count. Most mandatory values should be set either in the constructor or `build`.

Reply:
Agreed. Will fix this.